tests/test_random.py

```diff
@@ -287,6 +317,21 @@ def test_gaussian():
     np.testing.assert_almost_equal(var, v, 0,
             err_msg='Wrong variance from GaussianDeviate')
 
+    # Check discard
+    g2 = galsim.GaussianDeviate(testseed, mean=gMean, sigma=gSigma)
+    g2.discard(nvals)
+    v1,v2 = g(),g2()
+    print('after %d vals, next one is %s, %s'%(nvals,v1,v2))
+    assert v1 == v2
+    # Note: For Gaussian, this only works if nvals is even.
```

Comment:
Good idea.  I added a warning if either `rng.has_reliable_discard` is False or if an odd number is discarded and `rng.generates_in_pairs` is True.  This can be turned off using our regular `suppress_warnings` optional kwarg.